activities/ActivityTemplateVue.activity/js/activity.js

```diff
@@ -13,57 +13,50 @@ var app = new Vue({
 		currentUser: {
 			user: {}
 		},
-		presence: null,
-		journal: null,
-		localization: null,
-		icon: null,
+		SugarPresence: null,
+		SugarLocalization: null,
 		displayText: '',
 		pawns: [],
 		l10n: {
 			stringAddPawn: ''
 		}
 	},
 	mounted: function () {
-		this.presence = this.$refs.presence;
-		this.journal = this.$refs.journal;
-		this.localization = this.$refs.localization;
-
-		var vm = this;
-		requirejs(["sugar-web/activity/activity", "sugar-web/env", "sugar-web/graphics/icon"], function (activity, env, icon) {
-			// Initialize Sugarizer
-			activity.setup();
-
-			env.getEnvironment(function (err, environment) {
-				vm.currentenv = environment;
-				window.addEventListener('localized', initialLocalization);
-				function initialLocalization() {
-					vm.displayText = vm.localization.l10n.get("Hello", { name: environment.user.name });
-					vm.localization.localize(vm.l10n);
-					window.removeEventListener('localized', initialLocalization);
-				}
-			});
-
-			vm.icon = icon;
-		});
+		this.SugarPresence = this.$refs.SugarPresence;
+		this.SugarLocalization = this.$refs.SugarLocalization;
 	},
 
 	methods: {
 
+		initializeActivity: function() {
+			// Initialize Sugarizer
+			this.$refs.SugarActivity.setup();
+			this.currentenv = this.$refs.SugarActivity.getEnvironment();
+
+			var vm = this;
+			window.addEventListener('localized', initialLocalization);
+			function initialLocalization() {
+				vm.displayText = vm.SugarLocalization.l10n.get("Hello", { name: vm.currentenv.user.name });
+				vm.SugarLocalization.localize(vm.l10n);
+				window.removeEventListener('localized', initialLocalization);
```

Comment:
Not sure why you need to do that

Reply:
I created this to show the welcome message only once during the activity, and not be affected by any localized event. But I think the event is fired only once during the runtime of any activity. I'll add the message code to the localized method.

Reply:
Removed the `initialLocalization()` and made flow better in 06d7134.